chain/validation/types.go

```diff
@@ -0,0 +1,123 @@
+package validation
+
+import(
+	"fmt"
+	"github.com/pkg/errors"
+
+	"github.com/filecoin-project/chain-validation/pkg/state"
+	"github.com/filecoin-project/go-lotus/chain/address"
+	"github.com/filecoin-project/go-lotus/chain/types"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+//
+// Most of the file is based on the abi/ package from go-filecoin.
+//
+var ErrInvalidType = fmt.Errorf("invalid type")
+
+type Type uint64
+
+const (
+	// Invalid is the default value for 'Type' and represents an erroneously set type.
+	Invalid = Type(iota)
+	Address
+	AttoFIL
+	PeerID
+)
+
+type Value struct {
+	Type Type
+	Val  interface{}
+}
```

Comment:
I don't *think* we need this structure at all. Can we just reflect straight from a state-type into bytes?

Reply:
I think you are right, moved to here: https://github.com/filecoin-project/chain-validation/pull/10